core/indigo-core/common/base_cpp/scanner.cpp

```diff
@@ -796,6 +806,27 @@ byte BufferScanner::readByte()
     return _buffer[_offset++];
 }
 
+void BufferScanner::base64Decode(const std::string& in, std::string& out)
+{
+    std::vector<int> T(256, -1);
+    for (int i = 0; i < 64; i++)
+        T["ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789+/"[i]] = i;
+
+    int val = 0, valb = -8;
+    for (uint8_t c : in)
+    {
+        if (T[c] == -1)
+            break;
+        val = (val << 6) + T[c];
+        valb += 6;
+        if (valb >= 0)
+        {
+            out.push_back(char((val >> valb) & 0xFF));
+            valb -= 8;
+        }
+    }
+}
+
```

Comment:
There is no need in in this code, we already have cppcodec library in third_party directory, let's use it instead.

Reply:
fixed